mautic/mautic
# Fix campaign builder detach exception

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | staging
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #9056

#### Description:
Campaign event connections are not property removed because of missing `detach` funcion in `campaignBuilderInstance` object.  This builder issue may lead to campaign misconfiguration because it allows to set multiple event parents while only one is supported.

The `detach` function doesn't exist anymore it was replaced by `deleteConnection` (https://github.com/jsplumb/community-edition/issues/470).

```
TypeError: Mautic.campaignBuilderInstance.detach is not a function
    at _jp.Connection.<anonymous> (campaign.js:921)
    at Function.each (2.jquery.js:354)
    at Object.Mautic.campaignBeforeDropCallback [as beforeDrop] (campaign.js:920)
    at _jp.Endpoint.isDropAllowed (6.jsplumb.js:4101)
    at Object.isDropAllowed (6.jsplumb.js:8366)
    at Object.<anonymous> (6.jsplumb.js:8762)
    at Object.drop (6.jsplumb.js:3447)
    at Drop.drop (6.jsplumb.js:2368)
    at Drag.unmark (6.jsplumb.js:2216)
    at Drag.<anonymous> (6.jsplumb.js:2058)
```

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create campaign and try to assign multiple parents to one event. Every connection should be replaced by new one.
